### PR TITLE
Implicit transpose

### PR DESCRIPTION
**Pull request by zasdfgbnm**

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
!build

File: csrc/ir_interface_nodes.h [hunk 1]
```diff
@@ -544,6 +544,9 @@ class TORCH_CUDA_CU_API TensorView : public Val {
   // example, grouping multiple reductions.
   void updateMaxProducerPosition();
 
+  // commit the current changes in leaf domain into rFactor domain
```
Comment (naoyam) [inline csrc/ir_interface_nodes.h:547]: Can you expand the comment here as the usage of this function needs to be done very carefully. Some more info like intended usages would also be helpful.

File: csrc/tensor_view.cpp [hunk 2]
```diff
@@ -1476,6 +1476,21 @@ void TensorView::applyMmaSwizzle(MmaOptions options) {
   }
 }
 
+void TensorView::commitLeafToRFactor() {
+  TORCH_CHECK(
+      ir_utils::consumerTvsOf(this).empty(),
```
Comment (naoyam) [inline csrc/tensor_view.cpp:1481]: Can you test if this error check actually works?

The use list of each tensor is not going to be updated unless one of its indirect consumers is registered as a fusion output. So, for example:

```
tv0 = makeSymbolicTensor;
tv1 = set(tv0);
tv2 = sum(tv1);
tv1->split(...);
tv1->commitLeafToRFactor();
```

I think this won't results in an error as none is registered as fusion outputs yet.

I think it's ok to start with limited error check as I'm still not clear how this interface is going to be used.

**Review by naoyam — APPROVED** [review]
One concern on error check, but otherwise looks good

File: csrc/ir_interface_nodes.h (hunk 1 above)
Comment (zasdfgbnm) [reply csrc/ir_interface_nodes.h]: added

File: csrc/tensor_view.cpp (hunk 2 above)
Comment (zasdfgbnm) [reply csrc/tensor_view.cpp:1481]: Just checked. I think you are right. It won't raise an error unless you add output first...

**Comment by zasdfgbnm** [comment]
!build

**Comment by zasdfgbnm** [comment]
!build

**Pull request merged**